clients/src/main/java/org/apache/kafka/common/metrics/stats/Rate.java

```diff
@@ -59,10 +59,14 @@ public void record(MetricConfig config, double value, long timeMs) {
     @Override
     public double measure(MetricConfig config, long now) {
         double value = stat.measure(config, now);
+        return value / convert(windowSize(config, now));
+    }
+
+    public long windowSize(MetricConfig config, long now) {
         // the elapsed time is always N-1 complete windows plus whatever fraction of the final window is complete
-        long elapsedCurrentWindowMs = now - stat.current(now).lastWindowMs;
+        long elapsedCurrentWindowMs = (now - stat.oldest(now).lastWindowMs) % config.timeWindowMs();
```

Comment:
I second Joel's comment.. In particular, the difference in `lastWindowMs` of different samples is not necessarily multiple of config.timeWindowMs()